blockchain/chainio_test.go

```diff
@@ -449,88 +444,38 @@ func TestStxoDecodeErrors(t *testing.T) {
-			stxo:       spentTxOut{},
-			serialized: hexToBytes("4000016edbc6c4d31bae9f1ccc38538a114bf42de65e86"),
-			errType:    errDeserialize(""),
-			bytesRead:  23,
-		},
-		{
-			// [<flags 14> <script version 00> <compressed pk script 01 6e ...> <tx version 01> EOF]
-			name:       "no stakeextra data after script for a fully spent ticket stxo",
-			stxo:       spentTxOut{},
-			serialized: hexToBytes("4400016edbc6c4d31bae9f1ccc38538a114bf42de65e8601"),
-			errType:    errDeserialize(""),
-			bytesRead:  24,
-		},
-		{
-			// [<flags 14> <script version 00> <compressed pk script 01 6e ...> <tx version 01> <stakeextra {num outputs 01}> EOF]
-			name:       "truncated stakeextra data after script for a fully spent ticket stxo (num outputs only)",
-			stxo:       spentTxOut{},
-			serialized: hexToBytes("4400016edbc6c4d31bae9f1ccc38538a114bf42de65e860101"),
-			errType:    errDeserialize(""),
-			bytesRead:  25,
-		},
-		{
-			// [<flags 14> <script version 00> <compressed pk script 01 6e ...> <tx version 01> <stakeextra {num outputs 01} {amount 0f}> EOF]
-			name:       "truncated stakeextra data after script for a fully spent ticket stxo (num outputs and amount only)",
-			stxo:       spentTxOut{},
-			serialized: hexToBytes("4400016edbc6c4d31bae9f1ccc38538a114bf42de65e8601010f"),
-			errType:    errDeserialize(""),
-			bytesRead:  26,
-		},
-		{
-			// [<flags 14> <script version 00> <compressed pk script 01 6e ...> <tx version 01> <stakeextra {num outputs 01} {amount 0f} {script version 00}> EOF]
-			name:       "truncated stakeextra data after script for a fully spent ticket stxo (num outputs, amount, and script version only)",
-			stxo:       spentTxOut{},
-			serialized: hexToBytes("4400016edbc6c4d31bae9f1ccc38538a114bf42de65e8601010f00"),
-			errType:    errDeserialize(""),
-			bytesRead:  27,
-		},
-		{
-			// [<flags 14> <script version 00> <compressed pk script 01 6e ...> <tx version 01> <stakeextra {num outputs 01} {amount 0f} {script version 00} {script size 1a} {25 bytes of script instead of 26}> EOF]
-			name:       "truncated stakeextra data after script for a fully spent ticket stxo (script size specified as 0x1a, but only 0x19 bytes provided)",
-			stxo:       spentTxOut{},
-			serialized: hexToBytes("4400016edbc6c4d31bae9f1ccc38538a114bf42de65e8601010f001aba76a9140cdf9941c0c221243cb8672cd1ad2c4c0933850588"),
-			errType:    errDeserialize(""),
-			bytesRead:  28,
-		},
-	}
+	}{{
+		name:       "nothing serialized (no flags)",
+		stxo:       spentTxOut{},
+		txOutIndex: 0,
+		serialized: hexToBytes(""),
+		errType:    errDeserialize(""),
+		bytesRead:  0,
+	}, {
+		name:       "no data after flags",
+		stxo:       spentTxOut{},
+		txOutIndex: 0,
+		serialized: hexToBytes("01"),
+		errType:    errDeserialize(""),
+		bytesRead:  1,
+	}, {
+		name:       "incomplete compressed txout",
+		stxo:       spentTxOut{},
+		txOutIndex: 0,
+		serialized: hexToBytes("010012"),
+		errType:    errDeserialize(""),
+		bytesRead:  2,
+	}}
```

Comment:
Good call.  I updated these tests, as well as some of the other serialization/deserialization tests, to have more comprehensive coverage.  They now have pretty much full coverage, with the only missing cases being code paths that can't be hit (in the case of `decodeCompressedTxOut` it could be hit if called directly, but when called from the deserialize functions they already check for that case before calling it).
```
$ go test -run="TestStxoSerialization|TestStxoDecodeErrors|TestSpendJournalSerialization|TestSpendJournalErrors|TestUtxoSerialization|TestUtxoEntryDeserializeErrors" -coverprofile=cov.out
$ go tool cover -func=cov.out | grep -E "spentTxOutSerializeSize|putSpentTxOut|decodeSpentTxOut|deserializeSpendJournalEntry|serializeSpendJournalEntry|serializeUtxoEntry|deserializeUtxoEntry|encodeFlags|decodeFlags|compressedTxOutSize|putCompressedTxOut|decodeCompressedTxOut|readDeserializeSizeOfMinimalOutputs"
github.com/decred/dcrd/blockchain/v4/chainio.go:209:		readDeserializeSizeOfMinimalOutputs	100.0%
github.com/decred/dcrd/blockchain/v4/chainio.go:549:		spentTxOutSerializeSize			100.0%
github.com/decred/dcrd/blockchain/v4/chainio.go:569:		putSpentTxOut				100.0%
github.com/decred/dcrd/blockchain/v4/chainio.go:589:		decodeSpentTxOut			100.0%
github.com/decred/dcrd/blockchain/v4/chainio.go:642:		deserializeSpendJournalEntry		100.0%
github.com/decred/dcrd/blockchain/v4/chainio.go:704:		serializeSpendJournalEntry		93.8%
github.com/decred/dcrd/blockchain/v4/chainio.go:926:		serializeUtxoEntry			100.0%
github.com/decred/dcrd/blockchain/v4/chainio.go:964:		deserializeUtxoEntry			100.0%
github.com/decred/dcrd/blockchain/v4/compress.go:583:		compressedTxOutSize			100.0%
github.com/decred/dcrd/blockchain/v4/compress.go:601:		putCompressedTxOut			100.0%
github.com/decred/dcrd/blockchain/v4/compress.go:621:		decodeCompressedTxOut			86.4%
github.com/decred/dcrd/blockchain/v4/compress.go:698:		encodeFlags				100.0%
github.com/decred/dcrd/blockchain/v4/compress.go:714:		decodeFlags				100.0%
```